crates/katana/runner/src/lib.rs

```diff
@@ -157,3 +182,22 @@ impl Drop for KatanaRunner {
         }
     }
 }
+
+/// Determines the default program path for the katana runner.
+///
+/// If the CI environment variable is set, the default program path is "katana" to
+/// use the devcontainer version of katana.
+/// Otherwise, the default program path is determined by the KATANA_RUNNER_BIN environment variable
+/// that must be set by the developer.
+fn determine_default_program_path() -> String {
+    // GITHUB should define "CI". If not, we may use "GITHUB_ACTION".
+    match std::env::var("CI") {
+        Ok(_) => "katana".to_string(),
+        Err(_) => std::env::var("KATANA_RUNNER_BIN").unwrap_or_else(|_| {
+            panic!(
+                "KATANA_RUNNER_BIN environment variable is not set and required when not running \
+                 in CI."
+            )
+        }),
+    }
+}
```

Comment:
> i wonder if this would be consider as regression in testing DX... as now running tests locally would require everyone to build the latest katana, and for people that are working on katana, have to install the binary with their local changes in it.
> 
> i can imagine ppl getting confused when they're using outdated katana binary from `/.dojo` when they should've used the latest one installed using cargo in `/.cargo`

With the `TestSequencer`, katana was already re-built, but automatically that's for sure.
With the proposed solution here, it's only a matter of setting an env variable. Which is not that a big deal either. And you don't have to alter your path, which is less intrusive and heavy to setup. In VS studio (for people using it), it's also a one line to add to the `settings.json`.

> this makes me wonder, maybe the best way to separate the katana version from dojo is to just remove it from the workspace entirely (but still keep it in the monorepo)

Yep totally. And only the runner is exposed into `dojo-test-utils` or other crate, but which is not Katana crate.